src/briefcase/integrations/virtual_environment.py

```diff
@@ -0,0 +1,255 @@
+    """
+
+    def __init__(self, tools, venv_path: Path):
+        self.tools = tools
+        self.venv_path = venv_path
+
+    @property
+    def bin_dir(self) -> Path:
+        r"""Return the path to the virtual environment's binary directory.
+
+        :return: <venv>/bin (or <venv>\Scripts on Windows)
+        """
+        return self.venv_path / ("Scripts" if os.name == "nt" else "bin")
+
+    @property
+    def executable(self) -> str:
+        """Path to the Python executable in the virtual environment.
+
+        :return: Absolute filesystem path to the venv's Python executable.
+        """
+        python = self.bin_dir / ("python.exe" if os.name == "nt" else "python")
+        return os.fspath(python)
+
+    @property
+    def env(self) -> dict:
+        """Base environment variables for invoking processes in the venv.
+
+        The returned environment includes the venv's bin directory at the front of the
+        PATH, and sets VIRTUAL_ENV to the venv path. On Windows, PYTHONHOME is removed.
+        """
+
+        env = dict(os.environ)
+        old_path = env.get("PATH", "")
+        env["PATH"] = os.fspath(self.bin_dir) + (
+            os.pathsep + old_path if old_path else ""
+        )
+        env["VIRTUAL_ENV"] = os.fspath(self.venv_path)
+        if os.name == "nt":
+            env.pop("PYTHONHOME", None)
+        return env
+
+    def exists(self) -> bool:
+        """Check whether the virtual environment exists.
+
+        :return True if venv exists and contains a pyvenv.cfg file. else False
+        """
+        return self.venv_path.exists() and (self.venv_path / "pyvenv.cfg").exists()
+
+    def create(self) -> None:
+        """Create the virtual environment(if it doesn't already exist).
+
+        :raises BriefcaseCommandError if creation fails.
+        """
+        if not self.exists():
+            with self.console.wait_bar(
+                f"Creating virtual environment at {self.venv_path}..."
+            ):
+                try:
+                    self.venv_path.parent.mkdir(parents=True, exist_ok=True)
+                    self.tools.subprocess.run(
+                        [sys.executable, "-m", "venv", os.fspath(self.venv_path)],
+                        check=True,
+                    )
+                except subprocess.CalledProcessError as e:
+                    raise BriefcaseCommandError(
+                        f"Failed to create virtual environment at {self.venv_path}"
+                    ) from e
+
+    def recreate(self) -> None:
+        """Remove and re-create the virtual environment."""
+        if self.exists():
+            self.tools.console.info("Recreating virtual environment...")
+            shutil.rmtree(self.venv_path)
+        self.create()
+
+    def update_core_tools(self):
+        """Upgrade core Python packaging tooling in the venv."""
+        with self.tools.console.wait_bar(
+            "Upgrading pup tooling in virtual environment"
```

Comment:
```suggestion
            "Upgrading pip tooling in virtual environment"
```
but again, the output should be in the calling usage, if possible.